src/Altinn.App.Core/Features/Validation/ValidationService.cs

```diff
@@ -0,0 +1,156 @@
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Internal.AppModel;
+using Altinn.App.Core.Internal.Data;
+using Altinn.App.Core.Models.Validation;
+using Altinn.Platform.Storage.Interface.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.App.Core.Features.Validation;
+
+/// <summary>
+/// Main validation service that encapsulates all validation logic
+/// </summary>
+public class ValidationService : IValidationService
+{
+    private readonly IServiceProvider _serviceProvider;
```

Comment:
Using the ServiceProvider directly is required if we want to have support for Keyed services. Currently those lines adding that support are commented out, so currently it doesn't matter.

My opinion is that there isn't much practical difference between an ienumerable or a service provider, and we're still having a servicelocator pattern because we filter the ienumerable

Reply:
By injecting the ServiceProvider any service could be used in this service and as we aren't currently using keyedservices I think filtering IEnumerable of a given interface is a much more explicit way of doing it.